cnap/userv/pipeline_server.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/python3
+"""
+Pipeline Server provides a Restful API to get pipeline data.
+"""
+import os
+import sys
+import logging
+import signal
+from multiprocessing import Process
+
+from flask import Flask, jsonify, Response
+from flask_cors import cross_origin
+
+from core.rtdb import RedisDB, RuntimeDatabaseBase
+
+LOG = logging.getLogger(__name__)
+
+WEB_APP = Flask(__name__)
+server = Process(target=WEB_APP.run, args=('0.0.0.0',))
+
+def _get_env(key, default=None):
+    if key not in os.environ:
+        LOG.warning("Could not find the key %s in environment, "
+                    "use default value %s", key, str(default))
+        return default
+    return os.environ[key]
+
+class PipelineService:
+    """
+    Service to get pipeline data.
+    """
+    INFER_ENGINE_TABLE = "InferEngine-table"
+    PIPELINE_TABLE = "Pipeline-table"
+
+    _instance = None
+
+    def __init__(self):
+        self._db = None
+
+    @property
+    def db(self) -> RuntimeDatabaseBase:
+        """
+        Get the runtime database.
+
+        Returns:
+            RuntimeDatabaseBase: An instance of the RuntimeDatabaseBase class.
+        """
+        if self._db is None:
+            runtime_db_type = _get_env("RUNTIME_DB_TYPE", "redis")
+            if runtime_db_type == "redis":
+                self._db = RedisDB()
+                redis_host = _get_env("REDIS_HOST", "127.0.0.1")
+                redis_port = _get_env("REDIS_PORT", 6379)
+                self._db.connect(redis_host, redis_port)
+            else:
+                # TODO: support other runtime database type
+                LOG.error("Not supported runtime database type: %s", runtime_db_type)
+                raise NotImplementedError(f"Runtime database type {runtime_db_type} \
+                                          not supported yet.")
+        return self._db
+
+    @property
+    def pipelines(self) -> list:
+        """
+        Gets the pipeline list.
+
+        Returns:
+            list: List of pipeline dictionaries.
+        """
+        pipeline_dicts = self.db.get_all_table_objects_dict(PipelineService.PIPELINE_TABLE)
```

Comment:
Do we need to declare the exceptions that may be triggerred by `get_all_table_objects_dict()` function in the docstring?